src/renderer/component/publishForm/view.jsx

```diff
@@ -1,884 +1,574 @@
+      case 'other':
+        publishingLicense = otherLicenseDescription;
+        break;
+      default:
+        publishingLicense = licenseType;
     }
 
-    metadata.license = this.getLicense();
-    metadata.licenseUrl = this.getLicenseUrl();
-    metadata.nsfw = !!parseInt(this.state.meta_nsfw);
-
-    const doPublish = () => {
-      const publishArgs = {
-        name: this.state.name,
-        bid: parseFloat(this.state.bid),
-        metadata,
-        ...(this.state.channel != 'new' && this.state.channel != 'anonymous'
-          ? { channel_name: this.state.channel }
-          : {}),
-      };
-
-      const { source } = this.state;
-
-      if (this.refs.file.getValue() !== '') {
-        publishArgs.file_path = this.refs.file.getValue();
-      } else if (source) {
-        publishArgs.sources = source;
-      }
-
-      const success = claim => {};
-      const failure = error => this.handlePublishError(error);
-
-      this.handlePublishStarted();
-      this.props.publish(publishArgs).then(success, failure);
+    let publishingLicenseUrl = licenseType === 'copyright' ? '' : licenseUrl;
+
+    const publishParams = {
+      filePath,
+      bid,
+      title,
+      thumbnail,
+      description,
+      language,
+      nsfw,
+      channel,
+      license: publishingLicense,
+      licenseUrl: publishingLicenseUrl,
+      otherLicenseDescription,
+      copyrightNotice,
+      name,
+      contentIsFree,
+      price
     };
 
-    if (this.state.isFee && parseFloat(this.state.feeAmount) > 0) {
-      lbry.wallet_unused_address().then(address => {
-        metadata.fee = {
-          currency: this.state.feeCurrency,
-          amount: parseFloat(this.state.feeAmount),
-          address,
-        };
-
-        doPublish();
-      });
-    } else {
-      doPublish();
-    }
-  }
-
-  handlePublishStarted() {
-    this.setState({
-      modal: 'publishStarted',
-    });
-  }
-
-  handlePublishStartedConfirmed() {
-    this.props.navigate('/published');
-  }
-
-  handlePublishError(error) {
-    this.setState({
-      submitting: false,
-      modal: 'error',
-      errorMessage: error.message,
-    });
-  }
-
-  claim() {
-    const { claimsByUri } = this.props;
-    const { uri } = this.state;
-    return claimsByUri[uri];
-  }
-
-  topClaimValue() {
-    if (!this.claim()) return null;
-
-    return parseFloat(this.claim().effective_amount);
-  }
-
-  myClaimExists() {
-    const { myClaims } = this.props;
-    const { name } = this.state;
-
-    if (!name) return false;
-
-    return !!myClaims.find(claim => claim.name === name);
+    publish(publishParams);
   }
 
-  handleEditClaim() {
-    const claimInfo = this.claim() || this.myClaimInfo();
+  handleCancelPublish() {
+    const { clearPublish } = this.props;
 
-    if (claimInfo) {
-      this.handlePrefillClaim(claimInfo);
+    // #content wraps every page and lives in component/page
+    const mainContent = document.getElementById("content");
+    if (mainContent) {
```

Comment:
This creates a non-obvious dependency. I'd suggest making `scrollToTop` an action, at a minimum.